src/test/documentation_examples/mod.rs

```diff
@@ -1497,6 +1501,242 @@ async fn versioned_api_examples() -> std::result::Result<(), Box<dyn std::error:
     Ok(())
 }
 
+async fn aggregation_examples() -> GenericResult<()> {
+    let client = TestClient::new().await;
+    let db = client.database("aggregation_examples");
+    db.drop(None).await?;
+    aggregation_data::populate(&db).await?;
+
+    // Start Aggregation Example 1
+    let cursor = db
+        .collection::<Document>("sales")
+        .aggregate(
+            vec![
+                doc! { "$match" : { "items.fruit":"banana" } },
+                doc! { "$sort" : { "date" : 1 } },
+            ],
+            None,
+        )
+        .await?;
+    // End Aggregation Example 1
+    let out: Vec<_> = cursor.try_collect().await?;
+    assert_eq!(5, out.len());
+
+    // Start Aggregation Example 2
+    let cursor = db
+        .collection::<Document>("sales")
+        .aggregate(
+            vec![
+                doc! {
+                    "$unwind": "$items"
+                },
+                doc! { "$match": {
+                    "items.fruit" : "banana",
+                }},
+                doc! {
+                    "$group": {
+                        "_id": { "day": { "$dayOfWeek": "$date" } },
+                        "count": { "$sum": "$items.quantity" }
+                    }
+                },
+                doc! {
+                    "$project": {
+                        "dayOfWeek": "$_id.day",
+                        "numberSold": "$count",
+                        "_id":0
+                    }
+                },
+                doc! {
+                    "$sort": { "numberSold": 1 }
+                },
+            ],
+            None,
+        )
+        .await?;
+    // End Aggregation Example 2
+    let out: Vec<_> = cursor.try_collect().await?;
```

Comment:
Per [this comment](https://jira.mongodb.org/browse/DRIVERS-448?focusedCommentId=1810127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1810127), it looks like we should include iterating the cursor in the example too. As part of that we'll want to make sure to include a `use` for `TryStreamExt` within the example. Ditto for other examples.

Reply:
Updated to include the `use` and `try_collect` in the examples.  Gave each example its own scope to allow the otherwise-redundant `use` clauses, which is a bit of a hack but I didn't see a better way.